src/include/migraphx/module.hpp

```diff
@@ -109,21 +129,23 @@ struct module
 
     void finalize(context& ctx);
 
-    value to_value() const;
-    void from_value(const value& v);
-
     void debug_print() const;
     void debug_print(instruction_ref ins) const;
+    void debug_print(instruction_ref ins,
+                     const std::unordered_map<instruction_ref, std::string>& names) const;
     void debug_print(const std::vector<instruction_ref>& inss) const;
-    void print(const std::function<void(instruction_ref,
-                                        const std::unordered_map<instruction_ref, std::string>&)>&
-                   print_func) const;
+
+    std::unordered_map<instruction_ref, std::string> 
+        print(const std::function<void(instruction_ref, const std::unordered_map<instruction_ref, std::string>&)>&
+              print_func, std::unordered_map<instruction_ref, std::string> names) const;
```

Comment:
Names should be defaulted to an empty map.

Reply:
changed